src/config.rs

```diff
@@ -249,30 +273,84 @@ pub struct Config {
     pub directories: DirectoriesConfig,
 }
 
-impl From<RawConfig> for Config {
-    fn from(raw_config: RawConfig) -> Self {
-        Self {
-            style: StyleConfig {
-                command_name: raw_config.style.command_name.into(),
-                description: raw_config.style.description.into(),
-                example_text: raw_config.style.example_text.into(),
-                example_code: raw_config.style.example_code.into(),
-                example_variable: raw_config.style.example_variable.into(),
-            },
-            display: DisplayConfig {
-                compact: raw_config.display.compact,
-                use_pager: raw_config.display.use_pager,
-            },
-            updates: UpdatesConfig {
-                auto_update: raw_config.updates.auto_update,
-                auto_update_interval: Duration::from_secs(
-                    raw_config.updates.auto_update_interval_hours * 3600,
-                ),
-            },
-            directories: DirectoriesConfig {
-                custom_pages_dir: raw_config.directories.custom_pages_dir,
-            },
-        }
+impl TryFrom<RawConfig> for Config {
+    type Error = anyhow::Error;
+
+    /// Convert a `RawConfig` to a high-level `Config`.
+    ///
+    /// For this, some values need to be converted to other types and some
+    /// defaults need to be set.
+    fn try_from(raw_config: RawConfig) -> Result<Self, Self::Error> {
+        // Style config
+        let style = raw_config.style.into();
+
+        // Display config
+        let display = raw_config.display.into();
```

Comment:
Good call, here it absolutely makes sense to collapse the blocks, and it looks a bit silly otherwise 😄 It's a leftover from the refactoring / simplification.